pkg/discovery/globalnet/globalnet.go

```diff
@@ -284,38 +284,40 @@ func isCIDRPreConfigured(clusterID string, globalNetworks map[string]*GlobalNetw
 
 func ValidateGlobalnetConfiguration(subctlData *datafile.SubctlData, netconfig Config) (error, string, uint) {
 	status.Start("Validating Globalnet configurations")
+	var err error
+	var clusterSize uint
 	globalnetClusterSize := netconfig.GlobalnetClusterSize
 	globalnetCIDR := netconfig.GlobalnetCIDR
 	if subctlData.GlobalnetCidrRange != "" && globalnetClusterSize != 0 && globalnetClusterSize != subctlData.GlobalnetClusterSize {
-		clusterSize, err := GetValidClusterSize(subctlData.GlobalnetCidrRange, globalnetClusterSize)
+		clusterSize, err = GetValidClusterSize(subctlData.GlobalnetCidrRange, globalnetClusterSize)
 		if err != nil || clusterSize == 0 {
 			return fmt.Errorf("Invalid globalnet-cluster-size %s", err), "", 0
 		}
 		subctlData.GlobalnetClusterSize = clusterSize
 	}
 
 	if globalnetCIDR != "" && globalnetClusterSize != 0 {
-		err := errors.New("Both globalnet-cluster-size and globalnet-cidr can't be specified. Specify either one.\n")
+		err = errors.New("Both globalnet-cluster-size and globalnet-cidr can't be specified. Specify either one.\n")
 		return fmt.Errorf("%s", err), "", 0
 	}
 
 	if globalnetCIDR != "" {
-		_, _, err := net.ParseCIDR(globalnetCIDR)
+		_, _, err = net.ParseCIDR(globalnetCIDR)
 		if err != nil {
 			return fmt.Errorf("Specified globalnet-cidr is invalid: %s", err), globalnetCIDR, globalnetClusterSize
 		}
 	}
 
 	if subctlData.GlobalnetCidrRange == "" {
 		if globalnetCIDR != "" {
-			status.QueueSuccessMessage("globalnet is not enabled on Broker. Ignoring specified globalnet-cidr")
+			status.QueueWarningMessage("globalnet is not enabled on Broker. Ignoring specified globalnet-cidr")
 			globalnetCIDR = ""
 		} else if globalnetClusterSize != 0 {
-			status.QueueSuccessMessage("globalnet is not enabled on Broker. Ignoring specified globalnet-cluster-size")
+			status.QueueWarningMessage("globalnet is not enabled on Broker. Ignoring specified globalnet-cluster-size")
 			globalnetClusterSize = 0
 		}
 	}
-	status.End(true)
+	status.End(cli.CheckStatus(err))
```

Comment:
@mangelajo How about like this?

Reply:
@skitt like this?

Reply:
Nearly ;-). Your `CheckStatus` is fine, but our comments only concerned the replacement of `status.End(err == nil)`. Here you’re replacing `status.End(true)`; the replacement for that should be `status.End(success)` directly.

Reply:
How would status.End(0) (original implementation) different from status.End(success)? Just trying to understand. After all, its the same value in form of iota.

Reply:
From the computer’s perspective, the result is the same, as you say. However from a developer’s perspective it isn’t: if you read `status.End(0)`, you can’t know what it means without looking at the implementation of `status.End()`, which I’d rather avoid; if you read `status.End(status.Success)`, it’s somewhat easier to understand. Using symbolic names also allows us in the future to switch to a specific type, instead of generic `int`s, and thus constrain the allowable values. (As it is, one could write `success.End(4)` and the compiler wouldn’t complain.)